### PR TITLE
Failover gateway

### PR DESCRIPTION
**Pull request by EItanya**
# Description

Add helm values to configure Gloo as a failover target

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make update-deps generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3338

### CODE REVIEW

File: install/helm/gloo/templates/8-default-gateways.yaml [hunk 1]
```diff
@@ -64,5 +64,36 @@ spec:
...
+    tcpHosts:
+    - name: failover
+      sslConfig:
+        secretRef:
+          name: {{ $spec.failover.secretName }}
+          namespace: {{ $.Release.Namespace }}
+      destination:
+        forwardSniClusterName: {}
+  proxyNames:
+  - {{ $name | kebabcase }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/8-default-gateways.yaml:94]: this is a list that will always only have a single entry, is this intended?

File: install/test/helm_test.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"html/template"
 
+	v12 "github.com/solo-io/gloo/projects/gloo/pkg/api/v1"
```
Comment (kdorosh) [inline install/test/helm_test.go:9]: better import package name

File: changelog/v1.5.0-beta8/vs-status.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+changelog:
+  - type: FIX
+    issueLink: https://github.com/solo-io/gloo/issues/3115
+    description: >
+      Use the proxy status rather than entire proxy when calculating if we need to update the status
+      of a gateway resource. This change means we resync the status of gateway resources only when we need to.
+      When failing to write status, it will be retried a second later.
```
Comment (kdorosh) [inline changelog/v1.5.0-beta8/vs-status.yaml:7]: this is wrong

File: install/helm/gloo/templates/8-default-gateways.yaml (hunk 1 above)
Comment (EItanya) [reply install/helm/gloo/templates/8-default-gateways.yaml:94]: I am just copying the other 2 default gateways, this ensures that this Gateway only applies to this proxy

**Pull request merged**